src/GrantType/GrantTypeBase.php

```diff
@@ -63,12 +90,48 @@ protected function getAdditionalOptions()
     }
 
     /**
-     * @inheritdoc
+     * @param bool $forceCache
+     *
+     * @return AccessToken
      */
-    public function getToken()
+    public function getToken($forceCache = false)
     {
         $config = $this->config->toArray();
 
+        if ($this->cache) {
+            $key = $this->getCacheKey($config);
+
+            if ($forceCache || !$data = $this->cache->fetch($key)) { //cache miss
+
+                $lifetime = 0;
+                $data     = $this->getTokenDatas($config);
+                if (isset($data['expires'])) {
+                    $lifetime = (int) $data['expires'] - time();
+                    unset($data['expires']);
```

Comment:
You can store anything you like in `$data`: there is no need to unset() anything